baseplate/experiments/providers/__init__.py

```diff
@@ -103,20 +119,25 @@ def parse_experiment(config):
         override = config.get("global_override")
         return ForcedVariantExperiment(override)
 
-    if experiment_type == "r2":
-        return R2Experiment.from_dict(
+    experiment_class = type_class_map.get(experiment_type, ForcedVariantExperiment)
+
+    if experiment_type in ['r2', 'feature_flag']:
```

Comment:
it feels really weird to have the type map and then still branch on types. it'd be cleaner to move this branching into factory functions so this part of the code doesn't continue to need to be updated any time new variants are added.

Reply:
I've changed this a bit, so let me know if you still have objections here.  I'm not sure I'm entirely following what you're suggesting though; this, for all intents and purposes _is_ the factory function.

Keep in mind that the legacy types (r2, feature_flag, forced_variant) are slated for deprecation soon.